src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/advanced/GregtechMetaTileEntity_Adv_DistillationTower.java

```diff
@@ -364,29 +364,14 @@ public String getMachineType() {
     }
 
     @Override
-    public boolean checkRecipe(final ItemStack aStack) {
-        // Run standard recipe handling for distillery recipes
-        if (mMode == Mode.Distillery) {
-            return this.checkRecipeGeneric(getMaxParallelRecipes(), getEuDiscountForParallelism(), 100);
-        } else {
-            ItemStack[] inputs = getCompactedInputs();
-
-            for (GT_MetaTileEntity_Hatch_Input hatch : mInputHatches) {
-                FluidStack tFluid = hatch.getFluid();
-                if (tFluid != null) {
-                    if (checkRecipeGeneric(
-                            inputs,
-                            new FluidStack[] { tFluid },
-                            getMaxParallelRecipes(),
-                            100,
-                            250,
-                            10000)) {
-                        return true;
-                    }
-                }
-            }
-            return false;
-        }
+    protected ProcessingLogic createProcessingLogic() {
+        return new ProcessingLogic().setMaxParallelSupplier(this::getMaxParallelRecipes);
+    }
+
+    @Override
+    protected void setupProcessingLogic(ProcessingLogic logic) {
+        super.setupProcessingLogic(logic);
+        logic.setEuModifier(mMode == Mode.Distillery ? 0.15F : 1F);
     }
```

Comment:
Override this:
```java
@Override
public List<? extends IFluidStore> getFluidOutputSlots(FluidStack[] toOutput) {
    return getFluidOutputSlotsByLayer(toOutput, mOutputHatchesByLayer);
}
```
because this machine has special output hatch list

Reply:
The machine already had it https://github.com/GTNewHorizons/GTplusplus/blob/1e2dccfbf6ea2fe47595420599621e501570a2b3/src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/advanced/GregtechMetaTileEntity_Adv_DistillationTower.java#L356-L359

Reply:
oops